### ISSUE TITLE
encourage the use of nameof for constant string

### ISSUE THREAD

**Issue by fforjanAVEVA** [meziantou/Meziantou.Analyzer#1008]
**Is your feature request related to a problem? Please describe.**
As we want to avoid magic string, there is still time where the developer need to define common string.
We define them as constant to allow the reference management from compiler and IDE
But the question is always, what should be the constant name which always start with the same name as the constant.
```public const string Foo = "Foo";```

**Describe the solution you'd like**
Would like to have a new rule that detect  the constant matching the field name and emit the rule as the code should be written as 
```public const string Foo = nameof(Foo);```

The intent is in the future, if the string is change or the field is renamed but the string reverted as a string, we should catch it in the code review more easily.

PS: I understand this may be rejected for your package, we were thinking to implement in our own framework ruleset but wanted to see if the rule would be interesting at your level

**Comment by meziantou** [comment]
The rule itself could be valuable. However, it would provide false-positives if enabled for all string literals.

The current rules already suggest it for `ArgumentException` and nullable attributes if I'm not mistaken as the expected value is a symbol name. Also, there are existing rules 
- https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1507?WT.mc_id=DT-MVP-5003978
- https://github.com/dotnet/docs/blob/main/docs/fundamentals/code-analysis/style-rules/ide0280.md

I don't think I want to enable a general rule. However, for some specific usages where the intent of the string literal is to match a symbol name, then the rule can be valuable.

Next step: If you can find some clear usages, it can help shaping the rule.

**Comment by fforjanAVEVA** [comment]
In our case, the intent was only for constant field  ```public const string Foo = "Foo";```
We did not see any other usage at the moment as there is other rules

**Comment by meziantou** [comment]
The provided example doesn't seem to be a case where the literal is clearly expected to be a symbol name. Just matching a constant name may lead to false-positive. Maybe the way the value is used may provide more context.
